src/FSharpLint.Client/FSharpLintToolLocator.fs

```diff
@@ -0,0 +1,253 @@
+module FSharpLint.Client.FSharpLintToolLocator
+
+open System
+open System.ComponentModel
+open System.Diagnostics
+open System.IO
+open System.Text.RegularExpressions
+open System.Runtime.InteropServices
+open StreamJsonRpc
+open FSharpLint.Client.LSPFSharpLintServiceTypes
+
+let private supportedRange = SemanticVersioning.Range(">=v0.21.3") //TODO: proper version
+
+let private (|CompatibleVersion|_|) (version: string) =
+    match SemanticVersioning.Version.TryParse version with
+    | true, parsedVersion ->
+        if supportedRange.IsSatisfied(parsedVersion, includePrerelease = true) then
+            Some version
+        else
+            None
+    | _ -> None
+let [<Literal>] FSharpLintToolName = "dotnet-fsharplint"
+
+let private (|CompatibleToolName|_|) toolName =
+    if toolName = FSharpLintToolName then
+        Some toolName
+    else
+        None
+
+let private readOutputStreamAsLines (outputStream: StreamReader) : string list =
+    let rec readLines (outputStream: StreamReader) (continuation: string list -> string list) =
+        let nextLine = outputStream.ReadLine()
+
+        if isNull nextLine then
+            continuation List.Empty
+        else
+            readLines outputStream (fun lines -> nextLine :: lines |> continuation)
+
+    readLines outputStream id
+
+let private startProcess (ps: ProcessStartInfo) : Result<Process, ProcessStartError> =
+    try
+        Ok(Process.Start ps)
+    with
+    | :? Win32Exception as win32ex ->
+        let pathEnv = Environment.GetEnvironmentVariable "PATH"
+
+        Error(
+            ProcessStartError.ExecutableFileNotFound(
+                ps.FileName,
+                ps.Arguments,
+                ps.WorkingDirectory,
+                pathEnv,
+                win32ex.Message
+            )
+        )
+    | ex -> Error(ProcessStartError.UnexpectedException(ps.FileName, ps.Arguments, ex.Message))
+
+let private runToolListCmd (workingDir: Folder) (globalFlag: bool) : Result<string list, DotNetToolListError> =
+    let ps = ProcessStartInfo("dotnet")
+    ps.WorkingDirectory <- Folder.Unwrap workingDir
+    ps.EnvironmentVariables.["DOTNET_CLI_UI_LANGUAGE"] <- "en-us" //ensure we have predictible output for parsing
+
+    let toolArguments =
+        Option.ofObj (Environment.GetEnvironmentVariable "FSHARPLINT_SEARCH_PATH_OVERRIDE")
+        |> Option.map(fun env -> $" --tool-path %s{env}")
+        |> Option.defaultValue (if globalFlag then "--global" else String.Empty)
+
+    ps.CreateNoWindow <- true
+    ps.Arguments <- $"tool list %s{toolArguments}"
+    ps.RedirectStandardOutput <- true
+    ps.RedirectStandardError <- true
+    ps.UseShellExecute <- false
```

Comment:
@MrLuje let's use the ProcessStartInfo ctor that accepts all these params? this way you don't need to use the `<-` ugly mutation operator so many times here